cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/StepExecutionData.java

```diff
@@ -275,5 +286,30 @@ private String computeWorkerGroup(WorkerGroupStatement workerGroup,
         return workerGroupValue.toString();
     }
 
+    private void putStepNavigationOptions(RunEnvironment runEnv, Map<String, NavigationOptions> stepNavigationOptions) {
+        if (stepNavigationOptions != null) {
+            for (NavigationOptions options: stepNavigationOptions.values()) {
+                runEnv.putStepNavigationOptions(options.getCurrStepId(), stepNavigationOptions);
+                break;
+            }
+        }
+    }
+
+    private Double getRoiValue(String stepExecutableResult, Map<String, NavigationOptions> stepNavigationOptions) {
+        if (StringUtils.isNotEmpty(stepExecutableResult) && stepNavigationOptions != null) {
+            NavigationOptions navigationOptions = stepNavigationOptions.get(stepExecutableResult);
+            if (navigationOptions != null && navigationOptions.getOptions() != null) {
+                List<Map<String, Serializable>> options = navigationOptions.getOptions();
```

Comment:
why list of map here? why not Map or list of Pairs?

Reply:
Will rewrite this code